api-get.go

```diff
@@ -175,12 +154,12 @@ type objectReadSeeker struct {
 }
 
 // newObjectReadSeeker wraps getObject request returning a io.ReadSeeker.
-func newObjectReadSeeker(api API, bucket, object string) *objectReadSeeker {
+func newObjectReadSeeker(client Client, bucket, object string) *objectReadSeeker {
 	return &objectReadSeeker{
 		mutex:      new(sync.Mutex),
 		reader:     nil,
 		isRead:     false,
-		api:        api,
+		client:     client,
```

Comment:
This implementation is not right. Client should not be embedded inside readseeker. Client should feed readseeker via go routine.